syntax = "proto3";

package logger;

service LogService {
  rpc LogAction(LogRequest) returns (LogResponse) {}
}

enum Action {
  CREATE = 0;
  UPDATE = 1;
  DELETE = 2;
}

message LogRequest {
  int32 user_id = 1;
  Action action = 2;
  string entity = 3;
  string entity_id = 4;
  string changes = 5; // JSON string of changes
  string log_level = 6; // Optional: INFO, WARNING, ERROR, CRITICAL
}

message LogResponse {
  bool success = 1;
  string log_id = 2;
  string message = 3;
} 